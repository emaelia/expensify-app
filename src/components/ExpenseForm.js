import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';



//const now = moment();

//console.log(now.format('Do MMM YYYY'));

export default class ExpenseForm extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			description: props.expense ? props.expense.description  : '',
			note: props.expense ? props.expense.note : '',
			amount: props.expense ? (props.expense.amount / 100).toString() : '',
			createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
			calendarFocused: false,
			error: ''
		};
	}


	onDescriptionChange = (e) => {
		const description = e.target.value;
		this.setState(() => ({ description }));
	};

	onNoteChange = (e) => {
		const note = e.target.value;
		this.setState(() => ({ note }));
	};

	onAmountChange = (e) => {
		const amount = e.target.value;
		const regex = /^\d{1,}(\.\d{0,2})?$/;

		if(!amount || amount.match(regex)) {
			this.setState(() => ({ amount }))
		}
	};

	onDateChange = (createdAt) => {
		if(createdAt) {
			this.setState(() => ({ createdAt }));
		}
	};

	onFocusChange = ({ focused }) => {
		this.setState(() => ({ calendarFocused: focused }));
	};

	onSubmit = (e) => {
		e.preventDefault();

		if(!this.state.description || !this.state.amount) {

				this.setState(() => ({ error: 'Please provide the description and amount to proceed' }));

		} else {

			this.setState(() => ({ error: '' }));
			this.props.onSubmit ({
					description: this.state.description,
					amount: parseFloat(this.state.amount, 10) * 100,
					createdAt: this.state.createdAt.valueOf(),
					note: this.state.note
			});
		}
	};

	render() {
		return (
					
			<form onSubmit={this.onSubmit} className="form">
				{this.state.error && <p className="form__error">{this.state.error}</p> }
					<input 
						className="text-input"
						type='text' 
						placeholder='Description' 
						autoFocus
						value={this.state.description}
						onChange={this.onDescriptionChange}
					/>
					<input 
						className="text-input"
						type='number' 
						placeholder='Amount' 
						value={this.state.amount}
						onChange={this.onAmountChange}
					/>
					<SingleDatePicker
					 	date={this.state.createdAt} 
					 	onDateChange={this.onDateChange} 
					 	focused={this.state.calendarFocused} 
					 	onFocusChange={this.onFocusChange} 
						numberOfMonths={1}
						isOutsideRange={(day) => false}
					/>

					<textarea 
						className="textarea"
						placeholder='Add a note' 
						value={this.state.note}
						onChange={this.onNoteChange}
					></textarea>
					<div>
						<button className="button">Submit Expense</button>
					</div>
				
			</form>
			
		)
	}
};

