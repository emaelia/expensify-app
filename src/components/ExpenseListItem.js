import React from 'react';
import { Link } from 'react-router-dom'; 


const ExpenseListItem = ({ amount, description, createdAt, id }) => (
	<div className='expense'>
		<Link to={`/edit/${id}`} >
			<h3>
				{description}
			</h3>
		</Link>

		<p>
			Amount: {amount} - Date: {createdAt}
		</p>
	
	</div>
);



export default ExpenseListItem;