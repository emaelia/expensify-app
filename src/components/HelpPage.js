import React from 'react';
import { Link } from 'react-router-dom'; 

const HelpPage = () => (
	<div>How can I help you?</div>
	)

export default HelpPage;