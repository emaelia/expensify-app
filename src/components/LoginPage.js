import React from 'react';
import { connect } from 'react-redux';
import { startLogin, logout } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className='box-layout__box'>
            <h1 className='box-layout__title'>My Expenses</h1>
            <p>Keep your finances in check</p>
            <button onClick={startLogin} className='button' >Login with Google</button>

        </div>
     
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);