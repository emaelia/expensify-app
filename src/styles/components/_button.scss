.button {
    border: none;
    background-color: $matte-lipstick;
    display: inline-block;

    color: white;
    font-size: $font-size-large;
    font-weight: 300;
    text-decoration: none;
    line-height: 1;

    padding: $xs-size;
    margin: 0;
}

.button--link {
    background: none;
    border: none;
    padding: 0;
}

.button--secondary {
    background-color: $dull-gray;
}