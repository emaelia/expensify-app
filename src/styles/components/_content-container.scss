.content-container {

    margin: 0 auto;
    padding: 0 $xs-size 0 $s-size;
    max-width: 80rem;

}