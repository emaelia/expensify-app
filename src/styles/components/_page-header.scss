.page-header {
    background: $bright-gray;
    padding: $l-size 0;
    margin-bottom: $l-size;
}

.page-header__actions {
    margin-top: $m-size;
}

.page-header__title {
    margin: 0;

    font-size: $font-size-large;
    font-weight: 300;
    span {
        font-weight: 700;
    }
}