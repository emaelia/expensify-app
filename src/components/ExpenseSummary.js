import React from 'react';
import { connect } from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';
import getVisibleExpenses from '../selectors/expenses';
import ExpenseListItem from './ExpenseListItem';
import numeral from 'numeral';

export const ExpenseSummary = (props) => (
	<div>
			{
			props.expenses.length === 0 ? (
					<p>Total Pending</p>
				) : props.expenses === 1 ? (
					<p>Viewing {props.expenses} expense, which totals {numeral( props.expenseTotal / 100 ).format('$0,0.00')}</p>
				) : (
						
					<p>Viewing {props.expenses} expenses, which total {numeral( props.expenseTotal / 100 ).format('$0,0.00')}</p>
				)	
				
		}
	</div>	
);

const mapStateToProps = (state) => {

	const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);

return {
		expenses: visibleExpenses.length,
		expenseTotal: selectExpensesTotal(visibleExpenses)
	}	
};


export default connect(mapStateToProps)(ExpenseSummary);
