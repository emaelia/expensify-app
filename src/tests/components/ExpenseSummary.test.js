import { ExpenseSummary } from '../../components/ExpenseSummary';
import React from 'react';
import { shallow } from 'enzyme';


test('Should render ExpenseSummary correctly with one expense', () => {
	const wrapper = shallow(<ExpenseSummary expenses={1} expenseTotal={399} />);
	expect(wrapper).toMatchSnapshot()
});

test('Should render ExpenseSummary correctly with more than one expense', () => {
	const wrapper = shallow(<ExpenseSummary expenses={23} expenseTotal={213234399} />);
	expect(wrapper).toMatchSnapshot()
});