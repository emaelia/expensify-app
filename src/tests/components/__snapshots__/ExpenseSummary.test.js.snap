// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseSummary correctly with more than one expense 1`] = `
<div>
  <p>
    Viewing 
    23
     expenses, which total 
    $2,132,343.99
  </p>
</div>
`;

exports[`Should render ExpenseSummary correctly with one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     expense, which totals 
    $3.99
  </p>
</div>
`;
